### ISSUE TITLE
Decimals error?

### ISSUE THREAD

**Issue by trouble2** [nbrightproject/NBrightBuy#165]
I've entered 12,50 euro for some option, however in the cart it shows as 125,00.
In the totals it is calculating with the good value however:
So could something be wrong in:
@NBrightBuyUtils.FormatToStoreCurrency(nbiOpt.GetXmlPropertyDouble("option/optvalcost"))
![image](https://user-images.githubusercontent.com/1827559/111154080-0ecdff00-8593-11eb-9f81-65fa3f2e0939.png)

**Comment by leedavi** [comment]
Yes, looks like a display issue with the option price.  As a workaround for now I think you might be able to change the display format in the cart list template.

**Comment by trouble2** [comment]
And how would I do that?

**Comment by leedavi** [comment]
not sure of the code off the top of my head.... but change...

@NBrightBuyUtils.FormatToStoreCurrency(nbiOpt.GetXmlPropertyDouble("option/optvalcost"))

to something like this...

@nbiOpt.GetXmlPropertyDouble("option/optvalcost").ToString("C0")


Currency should be kept in a Decimal, legacy code has it as a double and it causes a lot of issues and time to get it correctly formatted.  But there are lots of formatting options https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings?redirectedfrom=MSDN#CFormatString

In OpenStore we have to support many currency formats, including the french who have a format "5,89" and decide to enter in the keypad "5.89".   We've had to do some JS and code to help these situations.  Maybe we're missing something on the money input field.

**Comment by trouble2** [comment]
Nope, that didn't solve it, there is def. something in the code that needs fixing, this is on the productdetail page:

![image](https://user-images.githubusercontent.com/1827559/111293507-a1c97080-8649-11eb-941d-e1a4ae113807.png)

And this is on the cartlist template:
![image](https://user-images.githubusercontent.com/1827559/111293677-cf161e80-8649-11eb-9cd8-b1ace058c4cc.png)

So whenever the amount is a whole number, like 35 everything is ok, when the amount is something like 12,50 it will show 125
Tried several things:
@nbiOpt.GetXmlPropertyDouble("option/optvalcost").ToString()
@nbiOpt.GetXmlProperty("option/optvalcost").ToString()
@nbiOpt.GetXmlPropertyDouble("option/optvalcost").ToString("F2")
@nbiOpt.GetXmlPropertyDouble("option/optvalcost").ToString("F")
@nbiOpt.GetXmlPropertyDouble("option/optvalcost").ToString("C0")

**Comment by trouble2** [comment]
The xml is correct though, and the calculations on the total also.
So it is something in the GetXMLProperty (Double) I guess....

`<optionvalues optionid="I35OCtZf">
		<genxml>
			<files />
			<hidden>
				<optionid>I35OCtZf</optionid>
				<optionvalueid>JMgf8npJ</optionvalueid>
			</hidden>
			<textbox>
				<txtaddedcost datatype="double">12.5</txtaddedcost>
			</textbox>
			<checkbox />
			<dropdownlist />
			<checkboxlist />
			<radiobuttonlist />
		</genxml>
	</optionvalues>
	<optionvalues optionid="lifbDOnk">
		<genxml>
			<files />
			<hidden>
				<optionid>lifbDOnk</optionid>
				<optionvalueid>DListlLm</optionvalueid>
			</hidden>
			<textbox>
				<txtaddedcost datatype="double">35</txtaddedcost>
			</textbox>
			<checkbox />
			<dropdownlist />
			<checkboxlist />
			<radiobuttonlist />
		</genxml>
	</optionvalues>`

**Comment by leedavi** [comment]
Here is my dev system... look OK???
![image](https://user-images.githubusercontent.com/5583141/111297020-70eb3a80-864d-11eb-9abb-1aa1e2482c66.png)

The is the default ClassicAjax Theme.

**Comment by leedavi** [comment]
![image](https://user-images.githubusercontent.com/5583141/111297289-be67a780-864d-11eb-8845-fd7c8b162405.png)

**Comment by trouble2** [comment]
Ok...so what could it be then? Is it the same code: 
@nbiOpt.GetXmlPropertyDouble("option/optvalcost")

Could it be some culture setting in DNN or something?

**Comment by leedavi** [comment]
This is in French... maybe something to do with language.  try a test system in French and see if it works.

**Comment by leedavi** [comment]
I think it is this...

            @if (@product.Options.Count >= 1)
            {
                <div class="options">@ProductOptions(product)</div>
            }

Not what you are formatting with.

**Comment by trouble2** [comment]
Yes, that is the part on the detailpage which is working fine.
But I mean in cartlist  which displays the 125 instead of 12,50

@NBrightBuyUtils.FormatToStoreCurrency(nbiOpt.GetXmlPropertyDouble("option/optvalcost"))

**Comment by trouble2** [comment]
Hmm....when I look in the order it shows this:
`<option>`
   `       <optid>FUTRWNRq</optid>`
    `      <optvalueid>True</optvalueid>`
    `      <optname>Verwijderingsbijdrage accu</optname>`
    `      <optvalcost datatype="double">125</optvalcost>`
     `     <optvaltext />`
    `      <optvaltotal datatype="double">12.5</optvaltotal>`
    `    </option>`

**Comment by trouble2** [comment]
See the difference between optvalcost and optvaltotal

**Comment by leedavi** [comment]
Ah!! Sorry, your using OpenStore v3 (NBrightBuy), I only just noticed.  Sorry we have stopped development on this.  I'm unsure of the fix, look at OpenStore for a solution, it seems to work in there. https://github.com/openstore-ecommerce/OpenStore

**Comment by trouble2** [comment]
Nope , we're using 4.1.4

**Comment by leedavi** [comment]
OK.  Here you put it on the v3 of OpenStore.  You should use this forum. 

https://github.com/openstore-ecommerce/OpenStore/discussions

But never mind.  The answer I put above is still valid if you use v4.  Have a look at the default templates and see what the difference is.

**Comment by trouble2** [comment]
Yes, well did you see the xml?
So when I change:
@NBrightBuyUtils.FormatToStoreCurrency((nbiOpt.GetXmlPropertyDouble("option/optval**cost**")))

to
@NBrightBuyUtils.FormatToStoreCurrency((nbiOpt.GetXmlPropertyDouble("option/optval**total**")))

it shows the correct 12,50 instead of 125

**Comment by leedavi** [comment]
Sorry, missed the email.

OK, so you think it should be "optvaltotal" for all templates?  It's strange it works for me. 
```
        <option>
          <optid>QEdTFc3V</optid>
          <optvalueid>True</optvalueid>
          <optname>2222222222</optname>
          <optvalcost datatype="double">2.3</optvalcost>
          <optvaltext>qqqq</optvaltext>
          <optvaltotal datatype="double">2.3</optvaltotal>
        </option>
```

I cannot remember the code for this changing for a long time.
What version is OpenStore ?
Did you import the data?
Try a few other tests and see if you can re-create the problem.
